spark-rapids/spark-rapids.sh

```diff
@@ -180,23 +177,83 @@ function install_nvidia_gpu_driver() {
+        Pin: release l=NVIDIA CUDA
+        Pin-Priority: 600
+        
+        Package: nvidia-settings
+        Pin: release l=NVIDIA CUDA
+        Pin-Priority: 600
+        
+        Package: *
+        Pin: release l=NVIDIA CUDA
+        Pin-Priority: 100
+        """ > /etc/apt/preferences.d/cuda-repository-pin-600
+        
+      apt install software-properties-common
+      
+      apt-key adv --fetch-keys ${NVIDIA_UBUNTU_REPO_URL}/3bf863cc.pub
+      add-apt-repository "deb ${NVIDIA_UBUNTU_REPO_URL} /"
+      
+      # CUDA_DRIVER_VERSION should be like "525.60.13-1"
+      CUDA_DRIVER_VERSION=$(apt-cache madison cuda-drivers | awk '{print $3}' | sort -r | while read line; do
+        if dpkg --compare-versions $(dpkg-query -f='${Version}\n' -W nvidia-driver-${NVIDIA_DRIVER_VERSION}) ge $line ; then
+           echo "$line"
+           break
+        fi
+      done)
+      
+#      apt-get install -y cuda-drivers-${NVIDIA_DRIVER_VERSION} cuda-drivers=${CUDA_DRIVER_VERSION}
+      apt install -y cuda-drivers-${NVIDIA_DRIVER_VERSION}=${CUDA_DRIVER_VERSION} cuda-drivers=${CUDA_DRIVER_VERSION}
+      
+      apt-get remove dkms && apt-mark hold dkms
+      
+      # the $line should be "cuda-runtime-12-0,cuda-drivers 525.85.12"
+      CUDA_VERSION=$(apt-cache showpkg cuda-drivers | grep -o 'cuda-runtime-[0-9][0-9]-[0-9],cuda-drivers [0-9\.]*' | while read line; do
+         if dpkg --compare-versions ${CUDA_DRIVER_VERSION} ge $(echo $line | grep -Eo '[[:digit:]]+\.[[:digit:]]+') ; then
+             echo $(echo $line | grep -Eo '[[:digit:]]+-[[:digit:]]')
+             break
+         fi
+      done)
+      
+      apt install -y cuda-${CUDA_VERSION}
+      
+    else
+      curl -fsSL --retry-connrefused --retry 3 --retry-max-time 5 \
       "${NVIDIA_UBUNTU_REPO_KEY_PACKAGE}" -o /tmp/cuda-keyring.deb
-    dpkg -i "/tmp/cuda-keyring.deb"
-    curl -fsSL --retry-connrefused --retry 3 --retry-max-time 5 \
-      "${NVIDIA_UBUNTU_REPO_CUDA_PIN}" -o /etc/apt/preferences.d/cuda-repository-pin-600
+      dpkg -i "/tmp/cuda-keyring.deb"
+      curl -fsSL --retry-connrefused --retry 3 --retry-max-time 5 \
+        "${NVIDIA_UBUNTU_REPO_CUDA_PIN}" -o /etc/apt/preferences.d/cuda-repository-pin-600
 
-    add-apt-repository "deb ${NVIDIA_UBUNTU_REPO_URL} /"
-    execute_with_retries "apt-get update"
+      add-apt-repository "deb ${NVIDIA_UBUNTU_REPO_URL} /"
+      execute_with_retries "apt-get update"
 
-    if [[ -n "${CUDA_VERSION}" ]]; then
-      local -r cuda_package=cuda-toolkit-${CUDA_VERSION//./-}
-    else
-      local -r cuda_package=cuda-toolkit
+      if [[ -n "${CUDA_VERSION}" ]]; then
+        local -r cuda_package=cuda-toolkit-${CUDA_VERSION//./-}
+      else
+        local -r cuda_package=cuda-toolkit
+      fi
+      # Without --no-install-recommends this takes a very long time.
+      execute_with_retries "apt-get install -y -q --no-install-recommends cuda-drivers-${DEFAULT_NVIDIA_DEBIAN_GPU_DRIVER_VERSION_PREFIX}"
```

Comment:
we can change this to ```execute_with_retries "apt-get install -y -q --no-install-recommends cuda-drivers```

Reply:
Hi @mengdong , since spark-rapids has released for about two weeks, the release work is included in this pr, we need to merge it ASAP, I will draft another pr to address the comments and reboot issues.